### PR TITLE
SCP-2427-3 Nerfs/Changes + Can_Rest system

### PR DESCRIPTION
**Pull request by Vizoplez**
## About the Pull Request
Bugfixes and adjusts 2427 to prevent future abuse and mechanical ignorance.

## Why It's Good For The Game
SCP-2427-3 has currently caused so many restarts/transfer votes due to buggy mechanics involving monkeys being unable to eaten, and in general, due to how powerful 2427-3 is; it has been a constant thorn in terms of server balance. This maintains its state as a powerful, tanky SCP; but nerfs most of the obnoxious gameplay loop it relied upon.

## Changelog

:cl:
tweak: 2427-3 now no longer counts monkeys as pure (should fix some edge cases)
fix: 2427-3 can no longer rest while alive. Prevents combat resting to avoid bullets.
tweak: 2427-3 higher hp loss from hunger. (0.01 -> 0.035)
tweak: 2427-3 HP and hit chance reduced & total damage output reduced (Fixes some dodge interactions + General nerf)
/:cl:

### CODE REVIEW

File: code/modules/mob/mob_defines.dm [hunk 1]
```diff
@@ -39,7 +39,6 @@
 	var/list/ooc_log = list()
 	var/list/dsay_log = list()
 	var/list/interact_log = list()
-
```
Comment (TheDarkElites) [inline code/modules/mob/mob_defines.dm]: dont change files unnecessarily, and before committing something to your pr double check, you didn't accidentally leave unnecessary changes, as you can just discard them at that stage instead of going back and fixing them.

File: code/modules/mob/living/living.dm [hunk 2]
```diff
@@ -721,11 +721,13 @@ default behaviour is:
 /mob/living/verb/lay_down()
 	set name = "Rest"
 	set category = "IC"
-
-	resting = !resting
-	to_chat(src, SPAN_NOTICE("You are now [resting ? "resting" : "getting up"]"))
-	if(hud_used)
-		hud_used.rest_button?.icon_state = "rest_[resting]"
+	if(src.can_rest)
```
Comment (TheDarkElites) [inline code/modules/mob/living/living.dm]:
```suggestion
	if(can_rest)
```
src. is assumed for any proc called inside of src

File: code/modules/SCP/SCPs/SCP-2427-3.dm [hunk 3]
```diff
@@ -10,10 +10,10 @@ GLOBAL_LIST_EMPTY(scp2427_3s)
 	attack_verb = list("stabbed")
 	hitsound = 'sound/scp/2427/stab.ogg'
 	damtype = BRUTE
-	melee_accuracy_bonus = 200
+	melee_accuracy_bonus = 60 //skull emoji
```
Comment (TheDarkElites) [inline code/modules/SCP/SCPs/SCP-2427-3.dm]: Please get rid of unnecessary comments, and remember that comments you make will be viewed without the context of what the pr changed, so anything relating to previous values should also not be included.

File: code/modules/SCP/SCPs/SCP-2427-3.dm [hunk 4]
```diff
@@ -206,6 +206,7 @@ GLOBAL_LIST_EMPTY(scp2427_3s)
 	return ..()
 
 /mob/living/simple_animal/hostile/scp_2427_3/SelfMove(direction)
+	resting = FALSE //2427 is forced to rest on dying... So make them not rest the next time they move.
```
Comment (TheDarkElites) [inline code/modules/SCP/SCPs/SCP-2427-3.dm:209]: leave this one though, this actually is a good comment to put

Comment (TheDarkElites) [reply code/modules/SCP/SCPs/SCP-2427-3.dm:209]: (you dont need to do anything for this just hit resolve on this one)

**Review by TheDarkElites — CHANGES_REQUESTED** [review]

File: code/modules/SCP/SCPs/SCP-2427-3.dm (hunk 4 above)
Comment (Tsurupeta) [reply code/modules/SCP/SCPs/SCP-2427-3.dm:209]: this looks stupid

**Review by Tsurupeta — DISMISSED** [review]
Remove useless comments and make it ignore resting state if can_rest == false in movement proc

File: code/modules/SCP/SCPs/SCP-2427-3.dm (hunk 4 above)
Comment (TheDarkElites) [reply code/modules/SCP/SCPs/SCP-2427-3.dm:209]: yeah it does dosent it

**Comment by Vizoplez** [comment]
> Remove useless comments and make it ignore resting state if can_rest == false in movement proc

This is unfortunately required even if it looks stupid; SCP-2427-3 can be forced to rest by things like flashbangs/etc due to horrible bay coding. This is just the most foolproof and bug-less way to ensure that they're force to stand on movement and attacking.

File: code/modules/SCP/SCPs/SCP-2427-3.dm (hunk 3 above)
Comment (Vizoplez) [reply code/modules/SCP/SCPs/SCP-2427-3.dm]: Fair enough.

File: code/modules/mob/living/living.dm (hunk 2 above)
Comment (Vizoplez) [reply code/modules/mob/living/living.dm]: Not...

neccessarily, but I'll commit and see where this goes :evil:

File: code/modules/mob/mob_defines.dm (hunk 1 above)
Comment (Vizoplez) [reply code/modules/mob/mob_defines.dm]: i have no idea why this changed, I didn't even touch this file.

Git-tortoise my beloved.

File: code/modules/SCP/SCPs/SCP-2427-3.dm [hunk 5]
```diff
@@ -39,15 +39,15 @@ GLOBAL_LIST_EMPTY(scp2427_3s)
...
+	can_rest = FALSE //Nope.
 	SCP = /datum/scp/scp_2427_3
 	status_flags = NO_ANTAG
 
 	see_invisible = SEE_INVISIBLE_NOLIGHTING
 	see_in_dark = 7
 
-	maxHealth = 1500
-	health = 1500
+	maxHealth = 1000 //500hp reduction.
```
Comment (TheDarkElites) [inline code/modules/SCP/SCPs/SCP-2427-3.dm]:
```suggestion
	maxHealth = 1000
```

File: code/modules/SCP/SCPs/SCP-2427-3.dm [hunk 6]
```diff
@@ -39,15 +39,15 @@ GLOBAL_LIST_EMPTY(scp2427_3s)
 	icon_dead = "dead"
 	pixel_x = -8
 	default_pixel_x = -8
-
+	can_rest = FALSE //Nope.
```
Comment (TheDarkElites) [inline code/modules/SCP/SCPs/SCP-2427-3.dm]:
```suggestion
	can_rest = FALSE
```

File: code/modules/SCP/SCPs/SCP-2427-3.dm [hunk 7]
```diff
@@ -117,7 +117,7 @@ GLOBAL_LIST_EMPTY(scp2427_3s)
 		CheckPurity(L)
 	AdjustSatiety(-satiety_reduction_per_tick)
 	if(satiety <= min_satiety) // Starvation, so you don't just run at mach 3 all the time
-		adjustBruteLoss(maxHealth * 0.01)
+		adjustBruteLoss(maxHealth * 0.06) //Starvation should kill you way faster.
```
Comment (TheDarkElites) [inline code/modules/SCP/SCPs/SCP-2427-3.dm]:
```suggestion
		adjustBruteLoss(maxHealth * 0.06)
```

File: code/modules/SCP/SCPs/SCP-2427-3.dm [hunk 8]
```diff
@@ -178,25 +178,25 @@ GLOBAL_LIST_EMPTY(scp2427_3s)
...
-		if((L.stat == DEAD) || (L.stat && ((L.health <= L.maxHealth * 0.25) || (L.getBruteLoss() >= L.maxHealth * 4))))
+		if((L.stat == DEAD) || L.isMonkey() || (L.stat && ((L.health <= L.maxHealth * 0.25) || (L.getBruteLoss() >= L.maxHealth * 4)))) //This is an informational comment; but please do not remove the 'ISMONKEY' check due to the fact that unless we recode the purity mechanism, 2427-3 Cannot attack them!
 			var/nutr = L.mob_size
 			if(istype(L, /mob/living/simple_animal/hostile/retaliate/goat)) // Likes goats
 				nutr = round(max_satiety * 0.5)
 			if(ishuman(L))
 				nutr = round(max_satiety * 0.2)
 			if(L.isMonkey())
-				nutr = round(max_satiety * 0.05)
+				nutr = round(max_satiety * 0.1) //Return To Monkey (Makes Monkey's half as filling as humans, instead of 0.01x)
```
Comment (TheDarkElites) [inline code/modules/SCP/SCPs/SCP-2427-3.dm]:
```suggestion
				nutr = round(max_satiety * 0.1)
```

**Review by TheDarkElites — CHANGES_REQUESTED** [review]

**Review by TheDarkElites — DISMISSED** [review]

**Comment by TheDarkElites** [comment]
putting in for TM

File: code/modules/SCP/SCPs/SCP-2427-3.dm [hunk 9]
```diff
@@ -117,7 +117,7 @@ GLOBAL_LIST_EMPTY(scp2427_3s)
 		CheckPurity(L)
 	AdjustSatiety(-satiety_reduction_per_tick)
 	if(satiety <= min_satiety) // Starvation, so you don't just run at mach 3 all the time
-		adjustBruteLoss(maxHealth * 0.01)
+		adjustBruteLoss(maxHealth * 0.06)
```
Comment (EgorDinamit) [inline code/modules/SCP/SCPs/SCP-2427-3.dm]: This has proven to be way too fast.
Settle for 1-3% instead.

File: code/modules/SCP/SCPs/SCP-2427-3.dm [hunk 10]
```diff
@@ -178,25 +178,25 @@ GLOBAL_LIST_EMPTY(scp2427_3s)
...
-		if((L.stat == DEAD) || (L.stat && ((L.health <= L.maxHealth * 0.25) || (L.getBruteLoss() >= L.maxHealth * 4))))
+		if((L.stat == DEAD) || L.isMonkey() || (L.stat && ((L.health <= L.maxHealth * 0.25) || (L.getBruteLoss() >= L.maxHealth * 4)))) //This is an informational comment; but please do not remove the 'ISMONKEY' check due to the fact that unless we recode the purity mechanism, 2427-3 Cannot attack them!
 			var/nutr = L.mob_size
 			if(istype(L, /mob/living/simple_animal/hostile/retaliate/goat)) // Likes goats
 				nutr = round(max_satiety * 0.5)
 			if(ishuman(L))
 				nutr = round(max_satiety * 0.2)
 			if(L.isMonkey())
-				nutr = round(max_satiety * 0.05)
+				nutr = round(max_satiety * 0.1)
```
Comment (EgorDinamit) [inline code/modules/SCP/SCPs/SCP-2427-3.dm]: No. This was done because monkey cubes cost extremely low; You are not meant to feed monkeys to it.

File: code/modules/mob/living/living.dm [hunk 11]
```diff
@@ -721,11 +721,13 @@ default behaviour is:
...
-	to_chat(src, SPAN_NOTICE("You are now [resting ? "resting" : "getting up"]"))
-	if(hud_used)
-		hud_used.rest_button?.icon_state = "rest_[resting]"
+	if(can_rest)
+		resting = !resting
+		to_chat(src, SPAN_NOTICE("You are now [resting ? "resting" : "getting up"]"))
+		if(hud_used)
+			hud_used.rest_button?.icon_state = "rest_[resting]"
+	else
+		to_chat(src, SPAN_NOTICE("You cannot rest!"))
```
Comment (EgorDinamit) [inline code/modules/mob/living/living.dm]:
```suggestion
	if(!can_rest)
		to_chat(src, SPAN_NOTICE("You cannot rest!"))
		return
	resting = !resting
	to_chat(src, SPAN_NOTICE("You are now [resting ? "resting" : "getting up"]"))
	if(hud_used)
		hud_used.rest_button?.icon_state = "rest_[resting]"
```

Comment (EgorDinamit) [reply code/modules/mob/living/living.dm]: This is preferable to the if-else kind of code writing.

File: code/modules/SCP/SCPs/SCP-2427-3.dm [hunk 12]
```diff
@@ -178,25 +178,25 @@ GLOBAL_LIST_EMPTY(scp2427_3s)
...
 		var/mob/living/carbon/human/H = A
-		if(H.stat != DEAD)
+		if(H.stat != DEAD && !H.isMonkey())
 			to_chat(src, SPAN_WARNING("We cannot decide if they are pure or not just yet..."))
 			return
 	if(isliving(A))
 		var/mob/living/L = A
 		// Brute loss part is mainly for humans
-		if((L.stat == DEAD) || (L.stat && ((L.health <= L.maxHealth * 0.25) || (L.getBruteLoss() >= L.maxHealth * 4))))
+		if((L.stat == DEAD) || L.isMonkey() || (L.stat && ((L.health <= L.maxHealth * 0.25) || (L.getBruteLoss() >= L.maxHealth * 4)))) //This is an informational comment; but please do not remove the 'ISMONKEY' check due to the fact that unless we recode the purity mechanism, 2427-3 Cannot attack them!
```
Comment (EgorDinamit) [inline code/modules/SCP/SCPs/SCP-2427-3.dm]: This makes it so 2427-3 instantly gibs all monkeys, regardless of their health and stat.
Really pointless change.

Comment (Vizoplez) [reply code/modules/SCP/SCPs/SCP-2427-3.dm]: Yes, because 2427-3 cannot eat monkeys that aren't dead currently due to how they index alive carbon mobs.

File: code/modules/SCP/SCPs/SCP-2427-3.dm (hunk 10 above)
Comment (Vizoplez) [reply code/modules/SCP/SCPs/SCP-2427-3.dm]: Fair.

File: code/modules/SCP/SCPs/SCP-2427-3.dm (hunk 9 above)
Comment (Vizoplez) [reply code/modules/SCP/SCPs/SCP-2427-3.dm]: > This has proven to be way too fast. Settle for 1-3% instead.

2427-3 should not breach for more than 5 minutes at a time ideally. It has repeatedly ended rounds by itself; but fair

Comment (EgorDinamit) [reply code/modules/SCP/SCPs/SCP-2427-3.dm]: 1% is exactly below 5 minutes. With 6% it dies in ~30 seconds.

File: code/modules/SCP/SCPs/SCP-2427-3.dm [hunk 13]
```diff
@@ -176,15 +176,15 @@ GLOBAL_LIST_EMPTY(scp2427_3s)
 	if(A in purity_list)
 		to_chat(src, SPAN_WARNING("They are pure... We will grant their wish."))
 		return
-	if(ishuman(A) && (satiety > min_satiety) && !(A in impurity_list))
+	if(ishuman(A) && (satiety > min_satiety) && !(A in impurity_list) || !isMonkey(A))
```
Comment (EgorDinamit) [inline code/modules/SCP/SCPs/SCP-2427-3.dm]: This will make it so 2427-3 cannot attack any humans, since they are not monkeys. Also, wrong proc call:
Should be `A.isMonkey()`. Either way, need to fix it.
```suggestion
	if(ishuman(A) && (satiety > min_satiety) && !(A in impurity_list) && !A.isMonkey())
```
I think it won't work, since A is an atom type and isMonkey is a mob/living proc, do with that what you will.

Comment (Vizoplez) [reply code/modules/SCP/SCPs/SCP-2427-3.dm]: Runtimes/Compile fails every other way, and by that logic wouldn't _(ishuman(A)_ fail?

Comment (EgorDinamit) [reply code/modules/SCP/SCPs/SCP-2427-3.dm]: ishuman is simply its own proc, while isMonkey, for some reason, is tied to mob.

**Review by EgorDinamit — APPROVED** [review]
I still have balance concerns, as this might make 2427-3 a bit too weak, but that can be adjusted later.
